src/installAddon.js

```diff
@@ -1,96 +1,89 @@
-import {parseDLURL_curseforge} from './curseforge'
 import fs from 'fs'
 import request from 'request-promise'
 import decompress from 'decompress'
 import { mainWindow } from '../main'
-const os = require('os');
-const path = require('path');
+import os from 'os'
+import path from 'path'
 
-// Create HTTP object
-export function makeHttpObject() {
-    try {return new XMLHttpRequest();}
-    catch (error) {}
-    try {return new ActiveXObject("Msxml2.XMLHTTP");}
-    catch (error) {}
-    try {return new ActiveXObject("Microsoft.XMLHTTP");}
-    catch (error) {}
-    throw new Error("Could not create HTTP request object.");
-}
+import { downloadUrlParserBuilder } from './pageParserAdapter/index'
 
 // Install the addon by chained-promises
-export function installAddon(addonObj, targetPath) {
-    return new Promise((resolve, reject) => {
+export function installAddon (addonObj, targetPath) {
```

Comment:
This looks great in the future we might wanna look into using the `await` operator in js to consolidate on the `.then`a

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await